### PR TITLE
Combobox: add select all functionality

### PR DESCRIPTION
**Pull request by smhigley**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Adds select all functionality to Combobox by adding `SelectAll` to ``SelectableOptionMenuItemType`. Adds controlled and uncontrolled examples of select all to react-examples.

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox.test.tsx [hunk 1]
```diff
@@ -794,6 +794,108 @@ describe('ComboBox', () => {
     });
   });
 
+  it('in multiSelect mode, selectAll selects all options', () => {
```
Comment (ecraig12345) [inline packages/react/src/components/ComboBox/ComboBox.test.tsx:797]: What happens if somebody includes a select all option to a ComboBox in single select mode? (Don't know why they would but...)

Comment (ecraig12345) [reply packages/react/src/components/ComboBox/ComboBox.test.tsx:797]: Also if they pass multiple select all options

**Review by ecraig12345 — APPROVED** [review]
LGTM, just might be good to have a couple more tests for the "I don't know why somebody would do this but when they do..." cases.

File: packages/react/src/components/ComboBox/ComboBox.test.tsx (hunk 1 above)
Comment (smhigley) [reply packages/react/src/components/ComboBox/ComboBox.test.tsx:807]: added a test! (the way I made it, it should act like a normal option 😄)

**Pull request merged**